File: Python/2025-RedistributeCharactersToMakeAllStringsEqual/2025-RedistributeCharactersToMakeAllStringsEqual.py
# Last updated: 25/09/2025, 08:27:12
class Solution:
    def makeEqual(self, words):
        n = len(words)
        freq = [0] * 26

        for word in words:
            for c in word:
                freq[ord(c) - ord('a')] += 1
        
        for count in freq:
            if count % n != 0:
                return False
        return True

File: 2025-RedistributeCharactersToMakeAllStringsEqual/2025-RedistributeCharactersToMakeAllStringsEqual.cs
// Last updated: 25/09/2025, 08:27:11
public class Solution {
    public bool MakeEqual(string[] words) {
        int n = words.Length;
        int[] freq = new int[26];

        foreach (var word in words){
            foreach (var c in word) {
                freq[c - 'a']++;
            }
        }

        foreach (var count in freq) {
            if (count % n != 0) return false;
        }
        return true;
    }
}